wallet-new/src/Cardano/Wallet/API/V1/Migration.hs

```diff
@@ -108,3 +111,60 @@ instance Migrate V0.SyncProgress V1.SyncProgress where
                 Just nd | _spLocalCD >= nd -> 100
                 Just nd -> round @Double $ (fromIntegral _spLocalCD / fromIntegral nd) * 100.0
         in pure $ V1.mkSyncProgress (fromIntegral percentage)
+
+
+--
+instance Migrate V0.CAccount V1.Account where
+    eitherMigrate V0.CAccount{..} =
+        V1.Account <$> eitherMigrate caId
+                  -- ^ accId
+                  <*> mapM eitherMigrate caAddresses
+                  -- ^ accAddresses
+                  <*> eitherMigrate caAmount
+                  -- ^ accAmount
+                  <*> pure (V0.caName caMeta)
+                  -- ^ accName
+                  <*> eitherMigrate caId
+                  -- ^ accWalletId
+
+instance Migrate V1.WalletId (V0.CId V0.Wal) where
+    eitherMigrate (V1.WalletId h) = pure $ V0.CId (V0.CHash h)
+
+--
+-- Following instances are friendly borrowed from @martoon's PR https://github.com/input-output-hk/cardano-sl/pull/2008
+-- TODO (jk): Use instances of his PR when it has been merged
+
+-- in old API 'V0.AccountId' supposed to carry both wallet id and derivation index
+instance Migrate (V1.WalletId, V1.AccountId) V0.AccountId where
```

Comment:
Interesting approach, probably we can't do better than that without complicating the types further.

Reply:
As mentioned in comment above those instances are `friendly borrowed` from @Martoon-00 's PR #2008 So kudos to Kostja :)